sentry-actix/src/lib.rs

```diff
@@ -299,3 +299,120 @@ fn process_event(
     }
     Some(event)
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io;
+
+    use actix_web::test::{call_service, init_service, TestRequest};
+    use actix_web::{get, web, App, HttpRequest, HttpResponse};
+
+    use sentry::{ClientInitGuard, Level};
+
+    use super::*;
+
+    fn _init() -> ClientInitGuard {
+        // Initialize with a 'valid' DSN so errors are captured
+        sentry::init("https://public@sentry.invalid/1")
+    }
+
+    fn _assert_hub_no_events() {
+        if Hub::current().last_event_id().is_some() {
+            panic!("Current hub should not have had any events.");
+        }
+    }
+
+    fn _assert_hub_has_events() {
+        Hub::current()
+            .last_event_id()
+            .expect("Current hub should have had events.");
+    }
+
+    /// Test explicit events sent to the current Hub inside an Actix service.
+    #[actix_rt::test]
+    async fn test_explicit_events() {
+        let _guard = _init();
+
+        let service = || {
+            // Current hub should have no events
+            _assert_hub_no_events();
+
+            // Add processor to ensure proper event fields
+            sentry::configure_scope(|scope| {
+                scope.add_event_processor(Box::new(move |event| {
+                    let request = event.request.expect("Request should be set.");
+                    assert_eq!(event.transaction, Some("/test".into()));
+                    assert_eq!(event.message, Some("Message".into()));
+                    assert_eq!(event.level, Level::Warning);
+                    assert_eq!(request.method, Some("GET".into()));
+                    None
```

Comment:
I’m a bit confused why you would get a last_event_id if you discard the event in the processor.

Reply:
Returning None discards the event, but it will still set the `last_event_id` to a `Default::default` Uuid in that case. It's a _bit_ of a hack, but it's the only way I could find to ensure the event was captured in the current Hub.

`sentry::test::with_captured_events` looks like a promising replacement, but the hurdle is that it creates a new Hub instance with an alternative transport to capture all events within it. It should still be feasible by passing the mock Hub to `SentryBuilder::with_hub`. I'll take a look!